kensu/nuclio/confluent_kafka/__init__.py

```diff
@@ -0,0 +1,27 @@
+from confluent_kafka import Producer as ConfluentProducer
+from kensu.nuclio.agent import kensu_add_kafka_output
+
+class Producer(ConfluentProducer):
+    def produce(self, topic, value=None, *args, **kwargs):
+        result = super().produce(topic, value, *args, **kwargs)
+        try:
+            if not self._kensu_nuclio_context:
+                return result
+            kensu_add_kafka_output(
+                context=self._kensu_nuclio_context,
+                topic=topic,
+                cluster_name=self._kensu_stored_config.get('bootstrap.servers') or 'unknown',
+                output_data=value)
+        except Exception:
+            pass
+        return result
+
+    def kensu_set_nuclio_context(self, context):
```

Comment:
or find a way so we don't need this.
if we assume one python interpreter always run only single function at the time (likely the case?), we could just store `lineage` in a variable  in package of `kensu.nuclio` instead of `context.user_data`.

Dunno if nuclio do any fail tolerance restarts and restoring/serializing user_data, would doubt about that. if not, we should be safe with variable in our package as described above, so we won't need to call this at all.

Reply:
should be hopefully solved by https://github.com/kensuio-oss/kensu-py/pull/179 @andypetrella